src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -67,12 +67,15 @@ protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descrip
             for (FoldingRange foldingRange : foldingRanges) {
                 TextRange textRange = getTextRange(foldingRange, file, document);
                 if ((textRange != null) && (textRange.getLength() > 0)) {
-                    String collapsedText = foldingRange.getCollapsedText();
-                    if (collapsedText != null) {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), textRange, null, collapsedText));
-                    } else {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), textRange));
-                    }
+                    descriptors.add(new FoldingDescriptor(
+                            root.getNode(),
+                            textRange,
+                            null,
+                            Collections.emptySet(),
+                            false,
+                            foldingRange.getCollapsedText(),
+                            (foldingRange instanceof LSPFoldingRange lspFoldingRange) && lspFoldingRange.isCollapsedByDefault()

```

Comment:
And now we just derive whether or not the range should be collapsed by default from the folding range itself as set, and only if it's our augmented subclass of `FoldingRange` with that information.